packages/react/src/dialog/Dialog.tsx

```diff
@@ -0,0 +1,50 @@
+import * as RadixDialog from "@radix-ui/react-dialog";
+import { type ComponentPropsWithRef, type ReactNode, forwardRef } from "react";
+
+import type { ExtendProps } from "../utils";
+
+import { Box } from "../box";
+import { Flex } from "../flex";
+import { CloseIcon } from "./CloseIcon";
+import * as styles from "./Dialog.css";
+
+type DialogProps = ExtendProps<
+  ComponentPropsWithRef<typeof Box>,
+  ComponentPropsWithRef<typeof RadixDialog.Root>,
+  {
+    children: ReactNode;
+    hideCloseIcon?: boolean;
+  } & styles.DialogVariants
+>;
+export const Dialog = forwardRef<HTMLDivElement, DialogProps>(
+  ({ children, hideCloseIcon = false, size = "md", ...props }, ref) => {
```

Comment:
technically you don't need to initialize falsy values
```suggestion
  ({ children, hideCloseIcon, size = "md", ...props }, ref) => {
```